### ISSUE TITLE
where and why to add padding

### ISSUE THREAD

**Issue by PedroDKE** [TensorSpeech/TensorFlowTTS#494]
so i have a quick and small question:

where do we add padding (eos tokens) in our sentences for training/evaluation? in the beginning, the end, somewhere in the middle or the end ?
And why do we add these extra tokens? is it in case that there is silence at the beginning/end of our training set or is there another reason?

also: are these padding tokens mapped one to one (in the mapper, is it for example '23:token' or is the token also split into different id's ?) and for the training of tacotron2, there is a stop token loss, do we need to specify the stop token somewhere or does the model already know this?

**Comment by OnceJune** [comment]
eos is short for [e]nd [o]f [s]entence, so the answer to your first question is, eos token should be added to the end of the input ling feat.
for your second question, why eos is necessary, in my understanding, its usage is to tell attention to give a stop token predict, since eos only occurs at the end of every training feat.
in nn networks, duration network has the knowledge that when the sent comes to the end. but attention in taco2 dose not know duration info, it just calc one alignment vector, sent to decoder, again and again. if you print the stop token sequence in every dynamic_decode step, you will find that for any non-eos token, value of stop token is quite small, but when come to eos token, the value of stop token comes to a quite large value, then the dynamic_decode know, oh, that's the end of this sent.

**Comment by PedroDKE** [comment]
@OnceJune thank you!
